pkgs/by-name/cm/cmake/package.nix

```diff
@@ -66,7 +67,13 @@ stdenv.mkDerivation (finalAttrs: {
   # Derived from https://github.com/curl/curl/commit/31f631a142d855f069242f3e0c643beec25d1b51
   ++ lib.optional (stdenv.isDarwin && isMinimalBuild) ./005-remove-systemconfiguration-dep.diff
   # On Darwin, always set CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG.
-  ++ lib.optional stdenv.isDarwin ./006-darwin-always-set-runtime-c-flag.diff;
+  ++ lib.optional stdenv.isDarwin ./006-darwin-always-set-runtime-c-flag.diff
+  # On platforms where ps is not part of stdenv, patch the invocation of ps to use an absolute path.
+  ++ lib.optional (stdenv.isDarwin || stdenv.isFreeBSD) (
+    substituteAll {
+      src = ./007-darwin-bsd-ps-abspath.diff;
+      ps = "${lib.getBin ps}/bin/ps";
+    });
```

Comment:
This might be more concisely done with a substituteInPlace or sed.

Reply:
I was under the impression that this was the kosher way to do this, since direct comparisons are hard to debug if the targeted code changes.

Reply:
Since the patch is conditional, doing it this way means that it's quite likely that Linux users will update the package, without realising that the patch no longer applies, meaning the package ends up not even building on FreeBSD and Darwin.

(OfBorg won't help here, because CMake updates have to go to staging, and OfBorg is generally useless on staging because it can never get past building LLVM without timing out.)

Reply:
isn't not building preferable to failing in mysterious ways down the line?

Reply:
I can also make it unconditional on linux, but this will be another mass rebuild.

Reply:
A mass rebuild will happen anyway in other machines.

Reply:
> isn't not building preferable to failing in mysterious ways down the line?

That can happen either way, if they start using it elsewhere in the file, for example.

> I can also make it unconditional on linux, but this will be another mass rebuild.

Probably best not to if it would bring in a dependency that doesn't get used.

Reply:
But I suppose even with sed or substituteInPlace we'd have the same risk of it not being caught due to having to do it conditionally, so I guess this is fine.